### PR TITLE
feat(seo): sensible meta descriptions for agency and post pages

### LINKED ISSUE #22: Replace `react-helmet` with `react-helmet-async`
`react-helmet` no longer works well with React 16, using the deprecated `UNSAFE_componentWillMount` (see nfl/react-helmet#548). The library is no longer maintained by the NFL, and has been dropped by the community in favour of react-helmet-async, written by @staylor of the NYT.

Work will chiefly centre around the following:
 https://github.com/opengovsg/askgovsg/blob/7e6350149167ce27820d089b7003a09cf6c9782b/client/src/components/PageTitle/PageTitle.component.jsx#L8-L10

### LINKED ISSUE #394: Populate sensible `<meta>` tags in questions and question lists
To allow crawlers to better understand and present AskGov information to users, embed `<meta>` tags and/or Open Graph tags using information that can be quickly obtained from the db.

You may wish to use Facebook's Sharing Debugger and similar tools to help determine effectiveness of your implementation.

### PR DESCRIPTION
**Pull request by LinHuiqing**
## Problem

Meta descriptions of dynamic pages currently do not reflect the content of the pages. An example is shown below:

Screenshot 2021-09-30 at 2 50 25 PM

Closes #394 and #22

## Solution

Dynamically populate meta descriptions for agency-specific pages to include the agency name and post pages with the answer to the question.

**Features**:

- update meta description for post page with text of first answer
- update meta description for agency-specific page with 'Answers from `agency.longname` (`agency.shortname`)' (e.g. 'Answers from Work Allocation Singapore (WAS)')
- set meta prop `data-rh` in `index.html` to `true`
- refactor AnswerSection component to prevent repeat answer queries

**Improvements**:

- refactor PageTitle component to typescript
- replace `react-helmet` with `react-helmet-async`

## Before & After Screenshots

**BEFORE**:

On agency-specific pages:

Screenshot 2021-09-30 at 3 18 02 PM

On post pages:

Screenshot 2021-09-30 at 3 17 48 PM

**AFTER**:

On agency-specific pages:

Screenshot 2021-09-30 at 3 13 21 PM

On post pages:

Screenshot 2021-09-30 at 3 13 51 PM

## Tests

1. Navigate to an agency-specific page and check the meta description with inspect - it should be 'Answers from `agency.longname` (`agency.shortname`)'.
2. Navigate to a post page and check the meta description with inpect - it should be the first answer to the question.
3. Navigate to the main AskGov page and check the meta description with inspect - it should be 'Answers from the Singapore Government'.

## Deploy Notes

**New dependencies**:

- `dependency`: 
  - `client/package.json`: `react-helmet-async`
  - `client/package.json`: `sanitize-html`
  - `client/package.json`: `@types/sanitize-html`

**Removed dependencies**:

- `dependency`: 
  - `client/package.json`: `react-helmet`

### CODE REVIEW

File: client/public/index.html [hunk 1]
```diff
@@ -35,6 +35,7 @@
     <meta
       name="description"
       content="Answers from the Singapore Government"
+      data-react-helmet="true"
```
Comment (LoneRifle) [inline client/public/index.html]: there's an outstanding issue to move to react-helmet-async. maybe we can move to that first?

File: client/src/pages/Post/Post.component.jsx [hunk 2]
```diff
@@ -69,12 +72,22 @@ const Post = () => {
...
+
   return isLoading ? (
     <Spinner centerHeight="200px" />
   ) : (
     <Flex direction="column" height="100%">
       <PageTitle
         title={`${post.title} - ${agencyShortName?.toUpperCase()} FAQ - AskGov`}
+        description={
+          answers && answers.length > 0
+            ? answers[0]?.body.replace(/<[^>]*>?/gm, '') // .replace() uses regex to remove html tags
```
Comment (LoneRifle) [inline client/src/pages/Post/Post.component.jsx]: can you use a sanitizer library -_-

**Review by LoneRifle — CHANGES_REQUESTED** [review]
looking good otherwise. You may want to consider populating other meta tags to improve our search ranking. eg, [keywords, Open Graph title(?)](https://gist.github.com/whitingx/3840905)

File: client/src/pages/Post/Post.component.jsx (hunk 2 above)
Comment (LinHuiqing) [reply client/src/pages/Post/Post.component.jsx]: Okay :') Actually JSX already escapes all the HTML tags so there isn't any security concern I think. But to be safe 👍🏻

Comment (LinHuiqing) [reply client/src/pages/Post/Post.component.jsx]: Hm for this, I could only find a package `html-to-text`, but am facing some issues with it due to the version of webpack (refer to https://github.com/html-to-text/node-html-to-text/issues/230). Will take a break from this first to look at the previous comment.

Comment (LoneRifle) [reply client/src/pages/Post/Post.component.jsx]: Ah I misread, you're just converting HTML to text for the description. okay, deal with react-helmet-async first =)

File: client/public/index.html (hunk 1 above)
Comment (LinHuiqing) [reply client/public/index.html]: Okay, issue https://github.com/opengovsg/askgovsg/issues/22 is addressed in https://github.com/opengovsg/askgovsg/pull/429/commits/115944fe2300f7f43b76cd04d3a2fd426520f1fe.

File: client/src/pages/Post/Post.component.jsx (hunk 2 above)
Comment (LinHuiqing) [reply client/src/pages/Post/Post.component.jsx]: Replaced with `sanitize-html` package as suggested.

**Comment by LinHuiqing** [comment]
@LoneRifle, regarding your comments on keywords and Open Graph titles, could we move it to a new issue or separate PR? I previously read that keywords might not be effective due to the common practice of keyword stuffing, but no one can say for certain as the crawlers' algorithms are secret.

**Review by LoneRifle — APPROVED** [review]
lgtm, good work! you can use Facebook's Sharing Debugger to figure out what else to populate by pointing it at staging, once this PR is deployed there.

**Pull request merged**